### ISSUE TITLE
Quoted command-line argument with space treats trailing backslash as escaping quote

### ISSUE THREAD

**Issue by TrueWill** [PowerShell/PowerShell#7400]
Related to https://github.com/PowerShell/PowerShell/issues/1995

Steps to reproduce
------------------
**foo.c:**
```
#include<stdio.h>
 
int main(int argc, char *argv[])
{
    if (argc >= 2)
        printf("%s", argv[1]);

    return 0;
}
```

Compiled with Microsoft C/C++ compiler v19.14.26433 for x86 to foo.exe

`./foo "C:\Program Files\"`

Expected behavior
-----------------
Output:
`C:\Program Files`

Actual behavior
---------------
Output:
`C:\Program Files"`

Environment data
----------------
```
Name                           Value
----                           -----
PSVersion                      5.1.17134.165
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17134.165
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by mklement0** [comment]
The issue you mention as related, #1995, is the catch-all issue for all problems relating to quoting of arguments for external programs.

The problem you describe here was previously reported in #4358, which was then closed and folded into #1995 via comment https://github.com/PowerShell/PowerShell/issues/1995#issuecomment-330675421

**Comment by mklement0** [comment]
I've just noticed that this bug is actually now fixed in PowerShell _Core_, on all platforms, as of at least PowerShell Core 6.1.0-preview.4, so I suggest closing this issue for that reason alone.

(However, other related problems mentioned in #1995 persist.)

Bugs that affect Windows PowerShell _only_ should be reported at [UserVoice](https://windowsserver.uservoice.com/forums/301869-powershell), which in this case has already happened: https://windowsserver.uservoice.com/forums/301869-powershell/suggestions/20102977-command-line-parsing-bug-spaces-and-trailing-slas
